### ISSUE TITLE
Joule fails to load due to refdes not being declared on some of the pins

### ISSUE THREAD

**Issue by jketreno** [Board-Explorer/board-viewer#38]
Load the [Joule](https://board-explorer.github.io/board-explorer/#joule_570x) and check the Javascript console. You will see:

![image](https://user-images.githubusercontent.com/742584/31407738-cf9790f2-adba-11e7-86b0-e7cbd33fd870.png)

**Comment by jketreno** [comment]
The above could be caused by the Joule not defining all the items fully in the board data file, however the parser should be catching that and either providing a default value, or providing a better error message.

**Comment by juliocri** [comment]
@jketreno I have been seeing around this and I've discovered that when a item is missing the ```item.refdes```  is deleted (line 1966), letting a incomplete structure, an object without refdes... and also if the missing components is intended to have pins, we let the pins in the object, i'm a litte confused in this point, due I found some ways to validate that, but I don't know what is the best fit:

1. at line 818: validate if the current index in the _```boardData``` have the property ```refdes```
2. deleting line 1966, in order to preserve the refdes, but in console the message shown will be something like the next:

```
Missing items:
{
  "J1": {
    "refdes": "J1",
    "pins": {
      "1": {},
      "2": {},
      "3": {},
      "4": {},
      "5": {},
      "6": {}
    }
  }
}
```
3. not deleting missing items.

what do you think?

**Comment by jketreno** [comment]
@juliocri Let's go with option #1 -- add a check to verify that _boardData...refdes exists.

**Comment by jketreno** [comment]
Published in 0.0.37
